src/Transport.ts

```diff
@@ -331,320 +338,383 @@ export default class Transport {
+            case 'TimeoutError':
+              if (!this[kRetryOnTimeout]) {
+                const wrappedError = new TimeoutError(error.message, result, errorOptions)
+                this[kDiagnostic].emit('response', wrappedError, result)
+                throw wrappedError
+              }
+            // should retry
+            // eslint-disable-next-line no-fallthrough
+            case 'ConnectionError': {
+              // if there is an error in the connection
+              // let's mark the connection as dead
+              this[kConnectionPool].markDead(meta.connection as Connection)
+
+              if (this[kSniffOnConnectionFault]) {
+                this.sniff({
+                  reason: Transport.sniffReasons.SNIFF_ON_CONNECTION_FAULT,
+                  requestId: meta.request.id,
+                  context: meta.context
+                })
               }
 
-              continue
+              // retry logic
+              if (meta.attempts < maxRetries) {
+                meta.attempts++
+                debug(`Retrying request, there are still ${maxRetries - meta.attempts} attempts`, params)
+
+                // exponential backoff on retries, with jitter
+                const backoff = options.retryBackoff ?? this[kRetryBackoff]
+                const backoffWait = backoff(0, 4, meta.attempts)
+                if (backoffWait > 0) {
+                  await setTimeoutPromise(backoffWait * 1000)
+                }
+
+                continue
+              }
+
+              // Wrap the error to get a clean stack trace
+              const wrappedError = error.name === 'TimeoutError'
+                ? new TimeoutError(error.message, result, errorOptions)
+                : new ConnectionError(error.message, result, errorOptions)
+              this[kDiagnostic].emit('response', wrappedError, result)
+              throw wrappedError
             }
 
-            // Wrap the error to get a clean stack trace
-            const wrappedError = error.name === 'TimeoutError'
-              ? new TimeoutError(error.message, result, errorOptions)
-              : new ConnectionError(error.message, result, errorOptions)
-            this[kDiagnostic].emit('response', wrappedError, result)
-            throw wrappedError
+            // edge cases, such as bad compression
+            default:
+              this[kDiagnostic].emit('response', error, result)
+              throw error
           }
-
-          // edge cases, such as bad compression
-          default:
-            this[kDiagnostic].emit('response', error, result)
-            throw error
         }
       }
+
+      return returnMeta ? result : result.body
     }
 
-    return returnMeta ? result : result.body
+    // wrap in OpenTelemetry span
+    if (params.meta?.name != null) {
+      return await otelTracer.startActiveSpan(params.meta.name, async (span) => {
```

Comment:
```suggestion
      return await otelTracer.startActiveSpan(params.meta.name, { kind: SpanKind.CLIENT }, async (span) => {
```

Also see the comment below about adding `attributes: ...` to the start span call.

Reply:
fixed in 264d99e232a967bd407b22cac571570f43df2219